### PR TITLE
[stable/k8s-spot-termination-handler] Add podAnnotations and updateStrategy features.

### PR DESCRIPTION
**Pull request by axozoid**
#### What this PR does / why we need it:
This PR brings 2 features:

1. ability to add `podAnnotations` thus making the config more flexible;
2. fixing a potential issue for people who had the app installed before Kubernetes 1.6 - when the default rolling update value was `OnDelete`;

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] title of the PR contains starts with chart name e.g. `[stable/chart]`

### CODE REVIEW

**Comment by axozoid** [comment]
@kierranm please have a look at my changes.

**Review by KierranM — APPROVED** [review]
Nice. Thanks for adding that, the changes are good. I don't actually have permission to merge or anything.

File: stable/k8s-spot-termination-handler/templates/daemonset.yaml [hunk 1]
```diff
@@ -65,3 +69,8 @@ spec:
     {{- if .Values.priorityClassName }} 
       priorityClassName: {{ .Values.priorityClassName }}
     {{- end }}
+  {{- if .Values.updateStrategy }}
+  updateStrategy:
+    type: {{ .Values.updateStrategy }}
```
Comment (cpanato) [inline stable/k8s-spot-termination-handler/templates/daemonset.yaml:74]: please add the options for the rolling update as well.
for example
```
    rollingUpdate:
      maxUnavailable: 1
```
```

Comment (axozoid) [reply stable/k8s-spot-termination-handler/templates/daemonset.yaml:74]: Done.

**Comment by axozoid** [comment]
@cpanato could you please review the PR?

**Comment by cpanato** [comment]
there is an error 
```
Error: release k8s-spot-termination-handler-ur7hxkufn7 failed: DaemonSet.apps "k8s-spot-termination-handler-ur7hxkufn7" is invalid: spec.updateStrategy.rollingUpdate.maxUnavailable: Invalid value: intstr.IntOrString{Type:1, IntVal:0, StrVal:"1"}: a valid percent string must be a numeric string followed by an ending '%!'(MISSING) (e.g. '1%!'(MISSING), or '93%!'(MISSING), regex used for validation is '[0-9]+%!'(MISSING))
```

@axozoid

**Comment by egeland** [comment]
I've fixed the issue in my rollup/refactor PR #14830

**Pull request closed without merging**